### PR TITLE
Remove tinygltf dependency from LL::GLTF

### PR DESCRIPTION
**Pull request by RunitaiLinden**

### CODE REVIEW

File: indra/newview/gltf/buffer_util.h [hunk 1]
```diff
@@ -353,33 +353,29 @@ namespace LL
             const Buffer& buffer = asset.mBuffers[bufferView.mBuffer];
             const U8* src = buffer.mData.data() + bufferView.mByteOffset + accessor.mByteOffset;
 
-            if (accessor.mComponentType == TINYGLTF_COMPONENT_TYPE_FLOAT)
+            if (accessor.mComponentType == Accessor::ComponentType::FLOAT)
```
Comment (akleshchev) [inline indra/newview/gltf/buffer_util.h:356]: I suggest using `switch` here

File: indra/newview/lllocalbitmaps.cpp [hunk 2]
```diff
@@ -1137,6 +1138,20 @@ void LLLocalBitmapMgr::delUnit(LLUUID tracking_id)
 	}
 }
 
+LLUUID LLLocalBitmapMgr::getTrackingID(const LLUUID& world_id) const
```
Comment (akleshchev) [inline indra/newview/lllocalbitmaps.cpp:1141]: Suggestion: implement `class LLViewerLocalTexture: public LLViewerFetchedTexture` that stores tracking id and replace `LLViewerFetchedTexture` in `LLLocalBitmap::updateSelf`
```
                    LLPointer<LLViewerFetchedTexture> texture = new LLViewerFetchedTexture
                        ("file://"+mFilename, FTT_LOCAL_FILE, mWorldID, LL_LOCAL_USE_MIPMAPS);
```
This way there won't be a need to search, just cast to LLViewerLocalTexture and get the tracking id.

File: indra/newview/gltf/asset.cpp [hunk 3]
```diff
@@ -1083,37 +890,60 @@ void Image::decompose(Asset& asset, const std::string& folder)
...
-        memcpy(data, mData.data(), mData.size());
-
-        LLViewerTextureList::createUploadFile(raw, filename, 4096);
+        auto bitmapmgr = LLLocalBitmapMgr::getInstance();
+        if (bitmapmgr->isLocal(mTexture->getID()))
+        {
+            LLUUID tracking_id = bitmapmgr->getTrackingID(mTexture->getID());
+            if (tracking_id.notNull())
+            { // copy original file to destination folder
+                std::string source = bitmapmgr->getFilename(tracking_id);
```
Comment (akleshchev) [inline indra/newview/gltf/asset.cpp:913]: You don't need `LLLocalBitmapMgr` for this. `mTexture` should have a specified url that is `"file://"+mFilename`

File: indra/newview/lllocalbitmaps.cpp (hunk 2 above)
Comment (RunitaiLinden) [reply indra/newview/lllocalbitmaps.cpp:1141]: I think I'd rather eat the search than add a sketchy typecast.  Once all this is functional it's going to get some heavy profiling over the summer and then we'll find out if the search is really relevant performance wise.

File: indra/newview/gltf/asset.cpp [hunk 4]
```diff
@@ -1023,13 +773,63 @@ void Asset::eraseBufferView(S32 bufferView)
...
+
+        U8* data = buffer.mData.data() + bufferView.mByteOffset;
+
+        mTexture = LLViewerTextureManager::getFetchedTextureFromMemory(data, bufferView.mByteLength, mMimeType);
+
+        if (mTexture.isNull())
+        {
+            LL_WARNS("GLTF") << "Failed to load image from buffer:" << LL_ENDL;
+            LL_WARNS("GLTF") << "  image: " << mName << LL_ENDL;
+            LL_WARNS("GLTF") << "  mimeType: " << mMimeType << LL_ENDL;
```
Comment (AndrewMeadows) [inline indra/newview/gltf/asset.cpp:801]: Really? Three **WARNING** lines instead of just one?

File: indra/newview/gltf/asset.cpp [hunk 5]
```diff
@@ -1083,37 +889,76 @@ void Image::decompose(Asset& asset, const std::string& folder)
...
+            error = "Unknown mime type, saved as .bin";
             extension = ".bin";
         }
 
-        std::string filename = folder + "/" + name + "." + extension;
+        std::string filename = folder + delim + name + extension;
 
         // set URI to non-j2c file for now, but later we'll want to reference the j2c hash
-        mUri = name + "." + extension;
+        mUri = name + extension;
```
Comment (AndrewMeadows) [inline indra/newview/gltf/asset.cpp:899]: `mUri = name + extension;`

So if you move that code above line 896 you could:

`std::string filename = folder + delim + mUri;`

**Review by AndrewMeadows — APPROVED** [review]
I made some comments and observations, but didn't spot any blockers.  Looks mergeable.

**Pull request merged**